ethereum/sidecar/server.go

```diff
@@ -53,12 +67,35 @@ var (
 	errInvalidEvents = fmt.Errorf("invalid AssetsLocked events")
 )
 
-// Server observes events emitted by the Mezo `MezoBridge` contract on the
-// Ethereum chain. It enables retrieval of information on the assets locked by
-// the contract. It is intended to be run as a separate process.
+// BridgeOutClient is a client enabling communication with the `Mezo` chain.
+type BridgeOutClient interface {
+	// GetAssetsUnlockedSequenceTip gets the assets unlocked sequence tip from
+	// the Mezo chain. The returned sequence tip is equal to the number of
+	// AssetsUnlocked events made so far. It is also equal to the value of the
+	// unlock sequence in the newest AssetsUnlocked event.
+	GetAssetsUnlockedSequenceTip(ctx context.Context) (sdkmath.Int, error)
+
+	// GetAssetsUnlockedEvents gets the AssetsUnlocked events from the Mezo
+	// chain. The requested range of events is inclusive on the lower side and
+	// exclusive on the upper side.
+	GetAssetsUnlockedEvents(
+		ctx context.Context,
+		sequenceStart sdkmath.Int,
+		sequenceEnd sdkmath.Int,
+	) ([]bridgetypes.AssetsUnlockedEvent, error)
+}
+
+// TimeFunc is a function that returns the current time.
+type TimeFunc func() time.Time
+
+// Server enables exchange of bridging-related information between the Mezo
+// and Ethereum chains. It used for both bridging-in of assets from Ethereum
+// to Mezo and bridging-out the other way around. It is intended to be run as
```

Comment:
nit: That could probably be reworded a little bit.

Reply:
Done in 656493ffbcd5198b430ff2ce59ffdfb7bf4a1d22.
Added more information to the description.